extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/execution/ExecutionConfig.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.k8s.overlord.execution;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+/**
+ * Represents the configuration for task execution within a Kubernetes environment.
+ * This interface allows for dynamic configuration of task execution strategies based
+ * on specified behavior strategies.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", defaultImpl = DefaultExecutionConfig.class)
+@JsonSubTypes(value = {
+    @JsonSubTypes.Type(name = "default", value = DefaultExecutionConfig.class)
+})
+public interface ExecutionConfig
+{
+  String CONFIG_KEY = "k8s.taskrunner.config";
```

Comment:
whats something that would go under this ExecutionConfig but not under ExecutionBehaviorStrategy? does it make more sense to call this KubernetesTaskRunnerRefreshableConfig or something like that?

Reply:
`ExecutionConfig` can have config other than `ExecutionBehaviorStrategy`, we may can move `RunnerStrategy` to this dynamic config (or something similar in the future):
```
{
  "type": "default",
  "behaviorStrategy": {
    "type": "default",
    "categorySelectors": [
    ]
  },
  "runnerStrategy": {
    ...
  }
}
```
In this case it's not only guide `KubernetesTaskRunner` behavior but also whether task should run in Worker, so I think the general ExecutionConfig name is making more sense.

Reply:
okay, ExecutionConfig makes sense to me then.

would categoryStrategy or templateStrategy make more sense for the second level thing then? since that config is really about choosing what template to run a task with?

Reply:
would task laning not be a separate field? like laneStrategy?

Reply:
I’m considering using a single strategy but with different fields returned. Since the implementation of the DynamicTaskExecutionBehaviorStrategy relies on the same Selector matching rules mechanism, the rules for matching categories and lanes are quite similar. Therefore, there’s no need to introduce a separate strategy for these functions.
```
public interface ExecutionBehaviorStrategy
{
  String getTaskCategory(Task task);
  String getTaskLane(Task task);
}
```

Reply:
YongGang and I discussed this offline, and I better understand the intent of this config object. It seems like this is trying to provide similar functionality as `KubernetesTaskRunnerConfig` but via the dynamic config. It makes sense to have an encompassing dynamic config object for this extension. Some suggested names for this that better indicate it's purpose `KubernetesTaskRunnerDynamicConfig` `KubernetesPeonDynamicConfig` `KubernetesTaskExecutionConfig`